internal/controller/instaslice_daemonset.go

```diff
@@ -939,3 +886,45 @@ func (r *InstaSliceDaemonsetReconciler) getInstasliceObject(ctx context.Context,
 
 	return &updateInstasliceObject, nil
 }
+
+func (r *InstaSliceDaemonsetReconciler) createSliceAndPopulateMigInfos(ctx context.Context, device nvml.Device, allocations inferencev1alpha1.AllocationDetails, giProfileInfo nvml.GpuInstanceProfileInfo, placement nvml.GpuInstancePlacement, ciProfileId int) (map[string]*MigDeviceInfo, error) {
+	log := logr.FromContext(ctx)
+
+	log.Info("creating slice for", "pod", allocations.PodName)
+
+	gi, ret := device.CreateGpuInstanceWithPlacement(&giProfileInfo, &placement)
+	if ret != nvml.SUCCESS {
+		if ret == nvml.ERROR_INSUFFICIENT_RESOURCES {
+			log.Error(ret, "GPU instance already exists")
+		} else {
+			log.Error(ret, "GPU instance creation errored out with unknown error")
+		}
+		return nil, fmt.Errorf("GPU instance creation failed: %v", ret)
+	}
+
+	giInfo, ret := gi.GetInfo()
+	if ret != nvml.SUCCESS {
+		log.Error(ret, "error getting GPU instance info", "giInfo", &giInfo)
+		return nil, fmt.Errorf("error getting GPU instance info: %v", ret)
+	}
+
+	ciProfileInfo, ret := gi.GetComputeInstanceProfileInfo(ciProfileId, 0)
+	if ret != nvml.SUCCESS {
+		log.Error(ret, "error getting compute instance profile info", "pod", allocations.PodName)
+		return nil, fmt.Errorf("error getting compute instance profile info: %v", ret)
+	}
+
+	ci, ret := gi.CreateComputeInstance(&ciProfileInfo)
+	if ret != nvml.SUCCESS {
+		log.Error(ret, "error creating Compute instance", "ci", ci)
+		return nil, fmt.Errorf("error creating compute instance: %v", ret)
+	}
+
+	migInfos, err := populateMigDeviceInfos(device)
+	if err != nil {
+		log.Error(err, "unable to iterate over newly created MIG devices")
+		return nil, fmt.Errorf("failed to populate MIG device infos: %v", err)
```

Comment:
s/%v/%w ?

Reply:
There are NVML return codes, there are tons of these and I don't know if they could be nil. I want to stick to %v for now.